inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -578,7 +578,7 @@ QueryNetworkResult clDNNEngine::QueryNetwork(const CNNNetwork& network,
             constantsNames.emplace(node->get_friendly_name());
             constants.push_back(node);
             return false;
-        } else if (prog.IsOpSupported(network, node) &&
+        } else if (prog.IsOpSupported(network, node, m_defaultContext->getImpl()->GetEngine()) &&
```

Comment:
I think this solution is overcomplicated. Why don't you just add new Program c-tor with single argument (engine), use it in line 529, and keep the rest of the code unchanged?